### PR TITLE
refactor:Refactor canvas component to utilize internal state [AB#17114]

### PR DESCRIPTION
**Pull request by tbarlow12**

### CODE REVIEW

File: src/react/components/pages/editorPage/canvas.test.tsx [hunk 1]
```diff
@@ -15,37 +15,41 @@ import { RegionsManager } from "vott-ct/lib/js/CanvasTools/Region/RegionsManager
...
+            selectedAsset: getAssetMetadata(),
+            onAssetMetadataChanged: jest.fn(),
             editorMode: EditorMode.Rectangle,
             selectionMode: SelectionMode.RECT,
             project: MockFactory.createTestProject(),
         };
 
         const assetPreviewProps: IAssetPreviewProps = {
-            asset: assetMetadata.asset,
+            asset: getAssetMetadata().asset,
```
Comment (elizabethhalper) [inline src/react/components/pages/editorPage/canvas.test.tsx]: Why in 44 can you just call "getAssetMetadata()" whereas here you use "getAssetMetadata().asset"?

File: src/react/components/pages/editorPage/canvas.test.tsx [hunk 2]
```diff
@@ -62,23 +66,22 @@ describe("Editor Canvas", () => {
...
-
     it("renders correctly from default state", () => {
+        const wrapper = createComponent();
+
         expect(wrapper.find(".canvas-enabled").exists()).toBe(true);
         expect(wrapper.state()).toEqual({
             contentSource: null,
             selectedRegions: [],
             canvasEnabled: true,
+            currentAsset: getAssetMetadata(),
```
Comment (elizabethhalper) [inline src/react/components/pages/editorPage/canvas.test.tsx]: (same as above comment)

File: src/react/components/pages/editorPage/canvas.test.tsx [hunk 3]
```diff
@@ -92,36 +95,50 @@ describe("Editor Canvas", () => {
...
     it("canvas is enabled when an asset is deactivated", () => {
+        const wrapper = createComponent();
         wrapper.find(AssetPreview).props().onDeactivated(expect.any(HTMLImageElement));
 
         expect(wrapper.instance().editor.addContentSource).toBeCalledWith(expect.any(HTMLImageElement));
         expect(wrapper.state().canvasEnabled).toEqual(true);
     });
 
     it("canvas is deactivated when an asset is activated", () => {
+        const wrapper = createComponent();
```
Comment (elizabethhalper) [inline src/react/components/pages/editorPage/canvas.test.tsx]: Can you call this line `const wrapper = createComponent();` in `beforeEach()`?

File: src/react/components/pages/editorPage/canvas.test.tsx [hunk 4]
```diff
@@ -139,65 +156,83 @@ describe("Editor Canvas", () => {
...
+    //     wrapper.state().currentAsset.regions.push(testRegion2);
+    //     canvas.editor.onRegionSelected("test1", false);
+    //     canvas.editor.onRegionSelected("test2", true);
+
+    //     const newTag = MockFactory.createTestTag();
+    //     canvas.onTagClicked(newTag);
+    //     for (const region of wrapper.instance().state.selectedRegions) {
+    //         expect(region.tags.findIndex((tag) => tag === newTag.name)).toBeGreaterThanOrEqual(0);
+    //     }
+    // });
```
Comment (elizabethhalper) [inline src/react/components/pages/editorPage/canvas.test.tsx]: Why is this test commented out?

File: src/react/components/pages/editorPage/canvas.tsx [hunk 5]
```diff
@@ -90,13 +93,9 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
         );
     }
 
-    /**
-     * Add tag to or remove tag from selected regions
-     * @param tag Tag to apply to or remove from selected regions
-     */
-    public onTagClicked = (tag: ITag) => {
+    public applyTag = (selectedTag: string) => {
```
Comment (elizabethhalper) [inline src/react/components/pages/editorPage/canvas.tsx]: Can you add JSDocs to this function?

File: src/react/components/pages/editorPage/editorPage.tsx [hunk 6]
```diff
@@ -181,30 +184,32 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
...
+        //                 new TagsDescriptor(selectedRegion.tags.map((tempTag) => new Tag(tempTag,
+        //                     this.props.project.tags.find((t) => t.name === tempTag).color))));
+        //         } else {
+        //             this.canvas.current.editor.RM.updateTagsById(selectedRegion.id, null);
+        //         }
+
+        //         return region;
+        //     });
+        // }
+        // this.onAssetMetadataChanged(selectedAsset);
```
Comment (elizabethhalper) [inline src/react/components/pages/editorPage/editorPage.tsx]: This should also be uncommented or deleted

File: src/react/components/pages/editorPage/canvas.test.tsx (hunk 1 above)
Comment (tbarlow12) [reply src/react/components/pages/editorPage/canvas.test.tsx]: Kind of a weird naming scheme... The `.asset` property refers to the actual `IAsset` while this object is the `IAssetMetadata`

File: src/react/components/pages/editorPage/canvas.test.tsx (hunk 3 above)
Comment (tbarlow12) [reply src/react/components/pages/editorPage/canvas.test.tsx]: I was doing that, but it had introduced a few race conditions in the expectations for each test

File: src/react/components/pages/editorPage/canvas.test.tsx (hunk 4 above)
Comment (tbarlow12) [reply src/react/components/pages/editorPage/canvas.test.tsx]: By mistake :) good catch

File: src/react/components/pages/editorPage/canvas.tsx (hunk 5 above)
Comment (tbarlow12) [reply src/react/components/pages/editorPage/canvas.tsx]: Good call

File: src/react/components/pages/editorPage/editorPage.tsx (hunk 6 above)
Comment (tbarlow12) [reply src/react/components/pages/editorPage/editorPage.tsx]: Thank you!

**Comment by pjlittle** [comment]
looks good - nice additional test.

**Pull request merged**